experimental/packages/opentelemetry-sdk-metrics-base/src/state/AsyncMetricStorage.ts

```diff
@@ -58,6 +58,16 @@ export class AsyncMetricStorage<T extends Maybe<Accumulation>> implements Metric
     this._deltaMetricStorage.batchCumulate(processed);
   }
 
+  private async _call(timeoutMillis: number) {
+    const observableResult = new ObservableResult();
+    let callPromise = Promise.resolve(this._callback(observableResult));
+    if (timeoutMillis > 0 && timeoutMillis !== Infinity) {
```

Comment:
I like this use of `Infinity`. 
I'm wondering if it would make sense to put this check into `callWithTimeout()`. :thinking: 

No need to do this in this PR, but if you let me know what you think I can do a little refactor and open a PR for it after this is merged. :slightly_smiling_face:

Reply:
Well... I'd find this can be a pitfall as it is not aligned with setTimeout behavior, so I've removed this part and determined the necessity of timeout by nullish check.

> https://nodejs.org/api/timers.html#settimeoutcallback-delay-args
> When delay is larger than 2147483647 or less than 1, the delay will be set to 1. Non-integer delays are truncated to an integer.